open_event/views/admin/home.py

```diff
@@ -6,14 +6,30 @@
 from flask_admin import expose
 from flask_admin.base import AdminIndexView
 from flask.ext.scrypt import generate_password_hash
+from datetime import datetime
+import jwt
 
 from ...helpers.data import DataManager, save_to_db,get_google_auth,get_facebook_auth
 from ...helpers.data_getter import DataGetter
 from ...helpers.helpers import send_email_after_account_create, send_email_with_reset_password_hash
-from open_event.models.user import User
 from open_event.helpers.oauth import OAuth, FbOAuth
 from flask import current_app
 
+def generate_token(user):
```

Comment:
This seems complicated. 
Maybe you can just modify the /api/v2/login endpoint to look for a `next` url parameter (query param) and make the login form send the request to /api/v2/login.
If the next query param exists, then the method for /api/v2/login can call this intended_url function which can redirect to desired url. If next query param doesn't exist, it can just return the normal access_token json.